### PR TITLE
Split long permission granting queries into chunks to avoid adb failures

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/8461

### CODE REVIEW

File: lib/tools/adb-commands.js [hunk 1]
```diff
@@ -97,13 +98,33 @@ methods.grantAllPermissions = async function (pkg, apk) {
...
+      }
+      cmdChunk = cmdChunk.concat(nextCmd);
+    }
+    if (cmdChunk.length) {
+      cmds.push(cmdChunk);
+    }
+    log.debug(`Got the following command chunks to execute: ${cmds}`);
+    let result = true;
+    for (let cmd of cmds) {
+      result = await this.shell(cmd) && result;
```
Comment (vrunoa) [inline lib/tools/adb-commands.js]: @mykola-mokhnach would it be good to know which permission failed, what do you think ?

Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: this will be visible anyway, because shell method throws an exception if the return code from adb is not equal to 0

Comment (vrunoa) [inline lib/tools/adb-commands.js]: oh ok. Nevermind then.

Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: I have added error handling there for a situation when adb returns non-zero code for one of chunks located closer to the head of the queue, but there are more at the tail. This gives a possibility to process the whole permissions queue and ignore the error on the higher level if necessary.

**Comment by imurchie** [comment]
Published in `appium-adb@2.23.0`

**Pull request merged**